components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/ExternalWorkflowTemplateImpl.java

```diff
@@ -0,0 +1,81 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.workflow.impl;
+
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.workflow.impl.bean.BPSProfile;
+import org.wso2.carbon.identity.workflow.impl.internal.WorkflowImplServiceDataHolder;
+import org.wso2.carbon.identity.workflow.mgt.bean.Parameter;
+import org.wso2.carbon.identity.workflow.mgt.bean.metadata.InputData;
+import org.wso2.carbon.identity.workflow.mgt.bean.metadata.Item;
+import org.wso2.carbon.identity.workflow.mgt.bean.metadata.MapType;
+import org.wso2.carbon.identity.workflow.mgt.bean.metadata.ParameterMetaData;
+import org.wso2.carbon.identity.workflow.mgt.exception.WorkflowException;
+import org.wso2.carbon.identity.workflow.mgt.workflow.AbstractWorkflow;
+import org.wso2.carbon.identity.workflow.mgt.workflow.TemplateInitializer;
+import org.wso2.carbon.identity.workflow.mgt.workflow.WorkFlowExecutor;
+
+import java.util.List;
+
+import static org.wso2.carbon.identity.workflow.impl.WFImplConstant.ParameterName.BPS_PROFILE;
+
+public class ExternalWorkflowTemplateImpl extends AbstractWorkflow {
+
+
+
+    public ExternalWorkflowTemplateImpl(Class<? extends TemplateInitializer> templateInitializerClass,
+                                        Class<? extends WorkFlowExecutor> workFlowExecutorClass, String metaDataXML) {
+        super(templateInitializerClass, workFlowExecutorClass, metaDataXML);
+    }
+
+    @Override
+    protected InputData getInputData(ParameterMetaData parameterMetaData) throws WorkflowException {
+        InputData inputData = null;
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        if (parameterMetaData != null && parameterMetaData.getName() != null) {
+            String parameterName = parameterMetaData.getName();
+            if (BPS_PROFILE.equals(parameterName)) {
+                List<BPSProfile> bpsProfiles = WorkflowImplServiceDataHolder.getInstance().getWorkflowImplService()
+                        .listBPSProfiles(tenantId);
+                if (bpsProfiles != null && bpsProfiles.size() > 0) {
+                    inputData = new InputData();
+                    MapType mapType = new MapType();
+                    inputData.setMapType(mapType);
+                    Item[] items = new Item[bpsProfiles.size()];
+                    for (int i = 0; i < bpsProfiles.size(); i++) {
+                        BPSProfile bpsProfile = bpsProfiles.get(i);
+                        Item item = new Item();
+                        item.setKey(bpsProfile.getProfileName());
+                        item.setValue(bpsProfile.getProfileName());
+                        items[i] = item;
+                    }
+                    mapType.setItem(items);
+                }
+            }
+        }
+        return inputData;
+    }
+
+    @Override
+    public void deploy(List<Parameter> parameterList) {
+
```

Comment:
Let's add a comment in the method body. No need to deploy as the external workflow is already deployed.

Reply:
comment addressed : https://github.com/wso2-extensions/identity-workflow-impl-bps/pull/71/commits/9822ae47e6a0e5c4b9b37b0ab00bd2a3b6fdca64